### PR TITLE
#3 Added aiSSEMBLE GitHub Action release job

### PR DESCRIPTION
**Pull request by jacksondelametter**
This is a PR to add the aiSSEMBLE GitHub Action release job. This action mimics the Jenkins release job found here https://github.boozallencsn.com/aissemble/aissemble/blob/fix-ci/devops/JenkinsfileRelease.groovy

The release job aims to do the following

- Parses the inputted version to get its major, minor, and patch counterparts
- Executes a pre release update to the release branch's Antora documentation 
- Cuts a release of the release branch
- Executes a post release update to the release branch's Antora documentation

### CODE REVIEW

File: .github/workflows/release.yml
Comment (jacksondelametter) [inline .github/workflows/release.yml:41]: Using env variables to easily switch out using workflow_dispatch input values and hardcoded values for testing!

File: .github/workflows/release.yml
Comment (jacksondelametter) [inline .github/workflows/release.yml:30]: Allows us to toggle running the release job in dryRun mode.

**Review by aaron-gary — APPROVED** [review]
Looks good to me, but I haven't done a release on aiSSEMBLE so be sure you get a second opinion from baseline.

File: .github/workflows/release.yml
Comment (carter-cundiff) [inline .github/workflows/release.yml:46]: A: Recommend making the env name ALL_CAPS if possible.

File: .github/workflows/release.yml
Comment (carter-cundiff) [inline .github/workflows/release.yml:87]: Q: `- uses: actions/checkout@v4` this is used twice, is that needed?

File: .github/actions/split_version/action.yaml
Comment (carter-cundiff) [inline .github/actions/split_version/action.yaml:14]: S: Would update this variable to be `major-minor-version`, sort of misleading currently.

File: .github/actions/split_version/action.yaml
Comment (carter-cundiff) [inline .github/actions/split_version/action.yaml:27]: S: Update descriptions

File: .github/actions/split_version/action.yaml
Comment (carter-cundiff) [inline .github/actions/split_version/action.yaml:45]: Q: Have you verified this works having the `echo` and part of the `&&` conditional?

Comment (jacksondelametter) [reply .github/actions/split_version/action.yaml:46]: Yes this works!

File: .github/workflows/release.yml
Comment (carter-cundiff) [inline .github/workflows/release.yml:123]: Q: Should we be using a differ pypi server for the release? I believe test pypi is only for our snapshots.

File: .github/workflows/release.yml
Comment (jacksondelametter) [reply .github/workflows/release.yml:89]: Yes it is needed, it is a bug with using custom actions. You have to checkout the current branch your running on for them to show up. The other `checkout` checks out the release branch.

File: .github/workflows/release.yml
Comment (jacksondelametter) [reply .github/workflows/release.yml]: Changed

File: .github/workflows/release.yml
Comment (carter-cundiff) [reply .github/workflows/release.yml:89]: Recommend grouping them together (if possible) and leaving a comment explaining that.

File: .github/workflows/release.yml
Comment (jacksondelametter) [reply .github/workflows/release.yml:123]: Its being uploaded to pypi not test pypi, I verified this.

File: .github/workflows/release.yml
Comment (jacksondelametter) [reply .github/workflows/release.yml:89]: I cant group them unfortunately but I can leave a comment

File: .github/workflows/release.yml
Comment (carter-cundiff) [reply .github/workflows/release.yml:123]: Safe to remove this then? `<id>dev-pypi</id><username>${{ secrets.TEST_PYPI_USERNAME }}</username><password>${{ secrets.TEST_PYPI_TOKEN }}</password>`?

Comment (jacksondelametter) [reply .github/workflows/release.yml:123]: I am not sure if it is safe to remove!

File: .github/actions/split_version/action.yaml
Comment (carter-cundiff) [inline .github/actions/split_version/action.yaml:42]: A: Would be helpful to have comments explaining what these vars are given inputs, for example from the old release script:
```
//grab second group (first embedded capture) in first match: 1.4.2-rc.3 -> 1.4.2

//grab second group (first embedded capture) in first match: 1.4.2 -> 1.4
```

File: .github/workflows/release.yml
Comment (carter-cundiff) [inline .github/workflows/release.yml:101]: Optional: Change `Development version` -> `Next Development Version` and `env.DEVELOPMENT_VERSION` -> `env.NEXT_DEVELOPMENT_VERSION`

File: .github/actions/split_version/action.yaml
Comment (jacksondelametter) [reply .github/actions/split_version/action.yaml:43]: Added comments to better explain

File: .github/workflows/release.yml
Comment (jacksondelametter) [reply .github/workflows/release.yml]: Changed

File: .github/workflows/release.yml
Comment (carter-cundiff) [inline .github/workflows/release.yml:36]: This will be removed before merge? Does it need a different value?

File: .github/workflows/release.yml
Comment (ewilkins-csi) [reply .github/workflows/release.yml:90]: I don't think you'll need a double checkout once this is merged to dev.  Do you have a link to the info you found saying you would need that?

File: .github/actions/split_version/action.yaml
Comment (ewilkins-csi) [inline .github/actions/split_version/action.yaml:24]: A: Update description

File: .github/actions/split_version/action.yaml
Comment (ewilkins-csi) [inline .github/actions/split_version/action.yaml:15]: S: In general these descriptions are just restating the name of the output which doesn't seem helpful.  We should either make them more explanatory or just omit them if we can.

File: .github/actions/update_antora/action.yaml
Comment (ewilkins-csi) [inline .github/actions/update_antora/action.yaml:19]: Q: Why are these inputs instead of just being derived from `version`?

File: .github/actions/update_antora/action.yaml
Comment (ewilkins-csi) [inline .github/actions/update_antora/action.yaml:42]: A: I'd make the commit message more accurate. This is likely a carryover from when we updated more stuff manually instead of through the build process.

File: .github/workflows/release.yml
Comment (ewilkins-csi) [reply .github/workflows/release.yml:36]: C: This _needs_ to be removed before merging.  This workflow should be on dispatch only.

File: .github/workflows/release.yml
Comment (ewilkins-csi) [inline .github/workflows/release.yml:48]: C: We should not be including these defaults at all. It just introduces risk/confusion.  If this script is updated/worked on in the future and such mechanisms are helpful, the author can add this stuff in.  However, I would argue that we should be testing by filling in the parameters manually anyway.

File: .github/workflows/release.yml
Comment (ewilkins-csi) [inline .github/workflows/release.yml:73]: S: This is just a copy of what we're doing in `build.yaml` right? Strongly suggest we pull these out into a common place if so.  We're already introducing the complexity of composite actions, and this is the core use case for them.

File: .github/workflows/release.yml
Comment (ewilkins-csi) [reply .github/workflows/release.yml:124]: yes it should be

File: .github/workflows/release.yml
Comment (ewilkins-csi) [inline .github/workflows/release.yml:127]: S: IMO the dry-run isn't going to be used enough to warrant having it checked in to dev and it just adds noise to the parameters. Esp since we aren't using it for things like the git commits at earlier steps.

File: .github/workflows/release.yml
Comment (jacksondelametter) [reply .github/workflows/release.yml]: Yes, this is only for testing purposes. It will be removed before being merged.

File: .github/actions/update_antora/action.yaml
Comment (jacksondelametter) [reply .github/actions/update_antora/action.yaml:19]: When updating the custom facet schemas, what we change differs between the pre release Antora docs update and the post release.

File: .github/workflows/release.yml
Comment (jacksondelametter) [reply .github/workflows/release.yml]: This is only for testing purposes. This will be deleted before being merged.

File: .github/workflows/release.yml
Comment (jacksondelametter) [reply .github/workflows/release.yml:54]: Ok this is actually not a bug. It looks like if you dont do a initial checkout, the local aissemble repository is actually not present in the runner. By doing an initial checkout, the repository is present and the branch being ran on is checked out.

File: .github/workflows/release.yml
Comment (jacksondelametter) [reply .github/workflows/release.yml]: Removed

File: .github/actions/split_version/action.yaml
Comment (jacksondelametter) [reply .github/actions/split_version/action.yaml]: Removed descriptions

File: .github/actions/update_antora/action.yaml
Comment (jacksondelametter) [reply .github/actions/update_antora/action.yaml]: Changed to "Updating Antora documentation to version ..."

File: .github/workflows/release.yml
Comment (jacksondelametter) [reply .github/workflows/release.yml]: Added instal dependencies composite action.

File: .github/workflows/release.yml
Comment (jacksondelametter) [reply .github/workflows/release.yml]: Removed

**Pull request merged**
